xlators/protocol/client/src/client-rpc-fops.c

```diff
@@ -3937,6 +3945,21 @@ client3_3_open(call_frame_t *frame, xlator_t *this, void *data)
 
     conf = this->private;
 
+    if (conf->strict_locks) {
+        pthread_spin_lock(&conf->fd_lock);
+        {
+            fdctx = this_fd_get_ctx(args->fd, this);
+            if (fdctx && !list_empty(&fdctx->lock_list)) {
+                ret = -1;
+                op_errno = EPERM;
```

Comment:
This should be EBADFD to be consistent.

Reply:
It is better to do this in client_open() so just doing once would be enough

Reply:
Done.